### PR TITLE
Swap out requests-async for httpx

### PR DESCRIPTION
**Pull request by ahopkins**

### CODE REVIEW

**Comment by ahopkins** [comment]
Making progress. Hope to have the last few tests passing by tomorrow.

**Comment by ahopkins** [comment]
Ready to go. @huge-success/sanic-release-managers

File: sanic/testing.py [hunk 1]
```diff
@@ -391,33 +215,39 @@ def __init__(
...
+        headers.setdefault("sec-websocket-key", "testserver==")
+        headers.setdefault("sec-websocket-version", "13")
+        if subprotocols is not None:
+            headers.setdefault(
+                "sec-websocket-protocol", ", ".join(subprotocols)
+            )
+
+        scope = {
+            "type": "websocket",
+            "asgi": {"version": "3.0"},
```
Comment (yunstanford) [inline sanic/testing.py]: should we expose the version or just hardcoded ?

File: tests/test_keep_alive_timeout.py [hunk 2]
```diff
@@ -1,15 +1,10 @@
...
 
 from asyncio import sleep as aio_sleep
-from http.client import _encode
 from json import JSONDecodeError
 
-import httpcore
-import requests_async as requests
-
-from httpcore import PoolTimeout
+# import httpcore
```
Comment (yunstanford) [inline tests/test_keep_alive_timeout.py]: nit: remove commented import.

File: sanic/testing.py (hunk 1 above)
Comment (ahopkins) [reply sanic/testing.py]: It really is not important for this purpose. This scope is merely a placeholder to allow the handshake and connection to be established. I think we are safe to leave it hardcoded.

File: tests/test_keep_alive_timeout.py (hunk 2 above)
Comment (ahopkins) [reply tests/test_keep_alive_timeout.py]: done

**Comment by ahopkins** [comment]
@sjsadowski I have some ideas in my head for making some enhancements to the testing  clients. But, I thought perhaps to save those for a separate PR.

**Pull request merged**